plugins/lookup/passwordstore.py

```diff
@@ -32,6 +32,12 @@
         description: Overwrite the password if it does already exist.
         type: bool
         default: 'no'
+      umask:
+        description: Sets the umask for the created .gpg files. The first octed must be greater than 3 (user readable).
+        default: '077'
```

Comment:
I'm wondering whether this really needs a default. How about simply not setting the env variable if a value is not provided?

The current behavior (enforcing this default if `PASSWORD_STORE_UMASK` is not set) could break backwards compatibility.

Reply:
I thought about this, too. What I was wondering was how have people in the past dealt with it and how might they have to have used the ``PASSWORD_STORE_UMASK`` setting in the past. 077 is the default (see ``man pass``). So either nobody cared, or they must have set the variable when executing the playbooks. 
Therefore I'd propose to either include a check as to whether the ``PASSWORD_STORE_UMASK`` is set in ``os.environ`` and issue a warning, or to remove the default from the ``self.paramvals`` as you suggested, such that backwards compatibility will be respected. 

I'd not remove this from the documentation, as it's otherwise unclear, what happns per default.

Reply:
I chose to remove the default setting in the plugin paramvals. Pass will take care of the rest and we don't have to check the environment variables if it's already set.

Reply:
You still have to remove the `default: '077'` here. This is code and documentation, not just documentation. You can specify in the description that the default is `'077'`, but not as `default:`.

(Right now the lookup is ignoring Ansible's parameter evaluation, but eventually when this changes, `default:` must **not** be specified here. And it's better to avoid it already now, before this creates a breaking change in the future when this plugin is fixed and the correct argument evaluation is used.)

Reply:
Interesting. I can't find any references to this in the documentation. Do you have any more info about these coming changes?

Reply:
They are not coming changes, they have been implemented some time ago. They're opt-in though, and this lookup hasn't chosen to use them yet. No idea where it's documented though.